src/constant.h

```diff
@@ -7,594 +7,598 @@
 // Global constants used by Moria
 // clang-format off
 
-#define MAX_UCHAR         255
-#define MAX_SHORT       32767   // maximum short/long signed ints
-#define MAX_LONG   0x7FFFFFFFL
+#pragma once
+
+#include "headers.h"
+
+constexpr int32_t MAX_UCHAR = std::numeric_limits<unsigned char>::max();
```

Comment:
I was looking over the code for usages of `MAX_UCHAR` and it seems it's only used for calculations not actual ASCII characters, so it would probably be better to use `uint8_t` instead of `unsigned char`.